src/components/navbars/DesktopNav.jsx

```diff
@@ -0,0 +1,35 @@
+import loginIcon from '../../assets/icons/login.png';
+import searchIcon from '../../assets/icons/search.png';
+
+const DesktopNav = ({ onSearch, onLogin, onJoin, stateSearch, stateLogin, stateJoin }) => {
+    return (
+        <nav className="w-2/3 hidden relative py-6 sm:flex justify-between">
+            {
+                stateSearch
+                    ? <form className="border-2 border-red rounded-full h-10 w-2/4 flex justify-between items-center px-4 ml-6 gap-1">
+                        <input className='focus:outline-none w-2/3' placeholder="Search online classes or teachers..." />
+                        <button onClick={onSearch}><img src={searchIcon} className="h-6" alt="search icon" /></button>
+                    </form>
+                    : <div></div>
+            }
+            <div id="mainNav" className="flex gap-4 items-center">
+                {
+                    stateLogin
+                        ? <button onClick={onLogin} className="font-spartan font-semibold text-base sm:text-lg flex items-center">
+                            <img src={loginIcon} className="h-4 sm:h-5" alt="join icon" />
+                            Log In
+                        </button>
+                        : <div></div>
```

Comment:
Instead of an empty `div` you can use fragments.

Reply:
got it

Reply:
without divs the buttons change their place on search page